pkg/restore/restore.go

```diff
@@ -474,14 +485,33 @@ func (ctx *context) restoreFromDir(dir string) (api.RestoreResult, api.RestoreRe
 			if !existingNamespaces.Has(mappedNsName) {
 				logger := ctx.log.WithField("namespace", nsName)
 				ns := getNamespace(logger, filepath.Join(dir, api.ResourcesDir, "namespaces", api.ClusterScopedDir, nsName+".json"), mappedNsName)
-				if _, err := kube.EnsureNamespaceExists(ns, ctx.namespaceClient); err != nil {
+
+				clusterNS, err := ctx.namespaceClient.Get(mappedNsName, metav1.GetOptions{})
```

Comment:
I think this part and the `ctx.createNamespaceAfterTerminate` part can be unified instead of being split:

```go
wait.PollImmediate(time.Second, ctx.resourceTerminatingTimeout, func() (bool, error) {
  clusterNS, err := ctx.namespaceClient.Get(mappedNsName, metav1.GetOptions{})
  if apierrors.IsNotFound(err) {
    if _, err := ctx.namespaceClient.Create(ns); err != nil {
      addArkError(&errs, err)
      return false, err
    }

    // ns created successfully; exit the loop
    return true, nil
  } else if err != nil {
    addArkError(&errs, err)
    return false, err
  }

  if clusterNS.Status.Phase == v1.NamespaceTerminating {
    // keep looping
    return false, nil
  }

  // namespace exists and is not terminating; exit the loop
  return true, nil
})